homeassistant/components/template/fan.py

```diff
@@ -340,23 +350,23 @@ def _update_state(self, result):
             self._state = None
             return
 
-        # Validate state
-        if result in _VALID_STATES:
+        if isinstance(result, bool):
             self._state = result
-        elif result in (STATE_UNAVAILABLE, STATE_UNKNOWN):
-            self._state = None
-        else:
-            _LOGGER.error(
-                "Received invalid fan is_on state: %s for entity %s. Expected: %s",
-                result,
-                self.entity_id,
-                ", ".join(_VALID_STATES),
-            )
-            self._state = None
```

Comment:
This change seems entirely unrelated to what's explained in the PR description?

Reply:
@emontnemery was out of town on business, I plan on tackling these comments soon.  Sorry for the delay.

Reply:
So, looking at this, I made these changes to make them more in line with the changes that occurred in a template switch because the template cover can now be stateless.  Covers in general could be stateless, but not template covers because of the assumed state in the background.  This change was added to accommodate that.

Reply:
I think you mean fan, not cover?

It's fine to update template fan to not only support on/off, but please move that to a separate PR which is not about making it possible to change set_percentage and set_preset_mode without implicitly turning the fan on? Or is there some reason why these changes must be done together?

Reply:
Yes, 🤣 I mean fans.

If we keep the old code we will be putting a string into the self._state field which is expecting True/False or None.  Not "on" or "off".   I can revert everything there, but there will still have to be changes to
```
        if result in _VALID_STATES:
            self._state = result
```
The old template fan, state was stored as a string.  The changes I made have it stored as a boolean that can be null.